Python/ceval.c

```diff
@@ -1924,6 +1926,87 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
             DISPATCH();
         }
 
+        case TARGET(BINARY_SUBSCR_ADAPTIVE): {
+            if (oparg == 0) {
+                PyObject *sub = TOP();
+                PyObject *container = SECOND();
+                next_instr--;
+                if (_Py_Specialize_BinarySubscr(container, sub, next_instr) < 0) {
+                    goto error;
+                }
+                DISPATCH();
+            }
+            else {
+                STAT_INC(BINARY_SUBSCR, deferred);
+                // oparg is the adaptive cache counter
+                *(next_instr-1) = _Py_MAKECODEUNIT(opcode, oparg-1);
```

Comment:
`opcode` might not be in scope here for some implementations of dispatch; use `BINARY_SUBSCR_ADAPTIVE` instead. Or `((uint8_t *)next_instr)[-1] = oparg-1` which is probably a bit more efficient.

Reply:
Is `((uint8_t *)next_instr)[-1] = oparg-1` correct also when `#ifdef WORDS_BIGENDIAN`?

Reply:
Yes, the layout is always opcode then oparg. 
We only need to know whether the platform is big endian or not, when decoding the `uint16_t` instruction.